BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -155,6 +155,10 @@ public class KeyringStore: ObservableObject {
         if self.selectedAccount.address != selectedAccountAddress {
           if let selectedAccount = selectedAccountKeyring.accountInfos.first(where: { $0.address == selectedAccountAddress }) {
             self.selectedAccount = selectedAccount
+          } else if let firstAccount = selectedAccountKeyring.accountInfos.first {
+            // try and correct invalid state (no selected account for this coin type)
+            self.selectedAccount = firstAccount
+            await self.keyringService.setSelectedAccount(firstAccount.address, coin: firstAccount.coin)
```

Comment:
Won't this call `setSelectedAccount` twice when `updateKeyringInfo` gets called outside of init? (Due to `selectedAccount.didSet`)

Reply:
Yes, nice catch, will remove the call to `keyringService`.

Reply:
Addressed in [40f5664](https://github.com/brave/brave-ios/pull/5845/commits/40f5664659ea1afa432738b4b3f0d844f6dada0c)